src/jit/lsrabuild.cpp

```diff
@@ -3054,22 +3116,64 @@ int LinearScan::BuildCmp(GenTree* tree)
 {
     assert(tree->OperIsCompare() || tree->OperIs(GT_CMP) || tree->OperIs(GT_JCMP));
     regMaskTP dstCandidates = RBM_NONE;
-    regMaskTP srcCandidates = RBM_NONE;
+    regMaskTP op1Candidates = RBM_NONE;
+    regMaskTP op2Candidates = RBM_NONE;
+    GenTree*  op1           = tree->gtGetOp1();
+    GenTree*  op2           = tree->gtGetOp2();
 
 #ifdef _TARGET_X86_
     // If the compare is used by a jump, we just need to set the condition codes. If not, then we need
     // to store the result into the low byte of a register, which requires the dst be a byteable register.
-    // We always set the dst candidates, though, because if this is compare is consumed by a jump, they
-    // won't be used. We might be able to use GTF_RELOP_JMP_USED to determine this case, but it's not clear
-    // that flag is maintained until this location (especially for decomposed long compares).
-    dstCandidates = RBM_BYTE_REGS;
-    if (varTypeIsByte(tree->gtOp.gtOp1) || varTypeIsByte(tree->gtOp.gtOp2))
+    if (tree->TypeGet() != TYP_VOID)
+    {
+        dstCandidates = RBM_BYTE_REGS;
+    }
+    bool needByteRegs = false;
+    if (varTypeIsByte(tree))
+    {
+        if (!varTypeIsFloating(op1))
+        {
+            needByteRegs = true;
+        }
+    }
+    // Example1: GT_EQ(int, op1 of type ubyte, op2 of type ubyte) - in this case codegen uses
+    // ubyte as the result of comparison and if the result needs to be materialized into a reg
+    // simply zero extend it to TYP_INT size.  Here is an example of generated code:
+    //         cmp dl, byte ptr[addr mode]
```

Comment:
The setcc insruction is missing from the example

Reply:
Also, I'm not sure why "in this case codegen uses ubyte as the result of comparison and if the result needs to be materialized into a reg simply zero extend it to TYP_INT size" appears here, this comment is more suitable where `dstCandidates` is set.